assets/js/modules/adsense/datastore/alerts.js

```diff
@@ -0,0 +1,209 @@
+		invariant( accountID, 'accountID is required.' );
+
+		return {
+			payload: { accountID, alerts },
+			type: RECEIVE_ALERTS,
+		};
+	},
+
+	receiveAlertsSucceeded( accountID ) {
+		invariant( accountID, 'accountID is required.' );
+
+		return {
+			payload: { accountID },
+			type: RECEIVE_ALERTS_SUCCEEDED,
+		};
+	},
+
+	receiveAlertsFailed( { accountID, error } ) {
+		invariant( accountID, 'accountID is required.' );
+		invariant( error, 'error is required.' );
+
+		return {
+			payload: { accountID, error },
+			type: RECEIVE_ALERTS_FAILED,
+		};
+	},
+};
+
+export const controls = {
+	[ FETCH_ALERTS ]: ( { payload: { accountID } } ) => {
+		return API.get( 'modules', 'adsense', 'alerts', { accountID } );
+	},
+};
+
+export const reducer = ( state, { type, payload } ) => {
+	switch ( type ) {
+		case FETCH_ALERTS: {
+			const { accountID } = payload;
+
+			return {
+				...state,
+				isFetchingAlerts: {
+					...state.isFetchingAlerts,
+					[ accountID ]: true,
+				},
+			};
+		}
+
+		case RECEIVE_ALERTS: {
+			const { accountID, alerts } = payload;
+
+			return {
+				...state,
+				alerts: {
+					...state.alerts,
+					[ accountID ]: [ ...alerts ],
+				},
+			};
+		}
+
+		case RECEIVE_ALERTS_SUCCEEDED: {
+			const { accountID } = payload;
+
+			return {
+				...state,
+				isFetchingAlerts: {
+					...state.isFetchingAlerts,
+					[ accountID ]: false,
+				},
+			};
+		}
+
+		case RECEIVE_ALERTS_FAILED: {
+			const { accountID, error } = payload;
+
+			return {
+				...state,
+				error,
+				isFetchingAlerts: {
+					...state.isFetchingAlerts,
+					[ accountID ]: false,
+				},
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getAlerts( accountID ) {
+		if ( ! isValidAccountID( accountID ) ) {
+			return undefined;
+		}
+		try {
+			const registry = yield Data.commonActions.getRegistry();
+			const existingAlerts = registry.select( STORE_NAME ).getAlerts( accountID );
+
+			// If there are already alerts loaded in state, consider it fulfilled
+			// and don't make an API request.
+			if ( existingAlerts ) {
+				return;
+			}
+
+			const alerts = yield actions.fetchAlerts( accountID );
+
+			yield actions.receiveAlerts( { accountID, alerts } );
+
+			return yield actions.receiveAlertsSucceeded( accountID );
+		} catch ( error ) {
+			// TODO: Implement an error handler store or some kind of centralized
+			// place for error dispatch...
+			return actions.receiveAlertsFailed( { accountID, error } );
+		}
+	},
+};
+
+export const selectors = {
+	/**
+	 * Gets all Google AdSense alerts for this account.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state     Data store's state.
+	 * @param {string} accountID The AdSense Account ID to fetch alerts for.
+	 * @return {?Array.<Object>} An array of AdSense alerts; `undefined` if not loaded.
+	 */
+	getAlerts( state, accountID ) {
+		invariant( accountID, 'accountID is required.' );
```

Comment:
As we've done elsewhere, we likely want to allow `undefined` here for selectors than may not have an `accountID` available yet. `null` shouldn't be allowed, because we should either pass a value here or just return `undefined` when `undefined` is supplied.

This is needed because our hooks need to run in a repeatable order inside components, so instead of doing:

```js
const accountID = useSelect(getAccountID);
if ( accountID ) {
  useSelect( select => select.getAlerts( accountID ) );
}
```

we have to allow for:

```js
const accountID = useSelect(getAccountID); // might return `undefined`
useSelect( select => select.getAlerts( accountID ) );
```

even if `accountID` is undefined.

Reply:
As discussed, let's add checks to all selectors that have required parameters:
* If any of the required parameters is `undefined`, immediately return `undefined`.
* That should be added to every such selector _and_ resolver.

cc @aaemnnosttv @tofumatt